projects/driver_controls/inc/magnetic_brake_event_generator.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "adc.h"
+#include "ads1015.h"
+#include "ads1015_def.h"
+
+typedef struct MagneticBrakeSettings {
+  int16_t percentage_threshold;  // value above which the brake_pressed event is raised
+  int16_t zero_value;            // lsb value obtained from ads1015 when brake is unpressed
+  int16_t hundred_value;         // lsb value obtained from ads105 when brake is pressed
+  int16_t min_allowed_range;     // min allowed percentage value, usually 0
+  int16_t max_allowed_range;     // max allowed percentage value, (1<<12)
+} MagneticBrakeSettings;
+
+typedef struct MagneticCalibrationData {
+  int16_t min_reading;
+  int16_t max_reading;  // value obtained from ads1015 in lsb
```

Comment:
This struct should just contain what this module needs to operate. Anything else should be stored in its own struct dedicated to calibration.

See: `throttle_calibration` has its own calibration storage and settings, but populates a calibration data struct from `throttle`.